modules/core/rapid_context.js

```diff
@@ -22,7 +22,7 @@ export function coreRapidContext() {
         var uniqueLngs = []; 
         if (gj.type === 'FeatureCollection') {
             var minlat, minlon, maxlat, maxlon;
-            //Calculate task extent. 
+            // Calculate task extent. 
             gj.features.forEach(function(f) {
```

Comment:
I logged `gj` in console and it looks like this:
![image](https://user-images.githubusercontent.com/7350578/65209959-93f05a80-da67-11e9-877e-04904b829a9b.png)

So `gj`'s "FeatureCollection" contains both the linestring for the bbox and all the points used by the linestring. This means that within the loop, we don't need to care about the `LineStrings` - looping through all the `Points` will already cover all coordinates used by the LingStrings. I think this is also the reason why my initial change didn't check against the LineString features. This will also make the logic simpler for cases where `gj` contains multiple `LineStrings`, e.g. this task contains 4 yellow polygons https://tasks-assisted.hotosm.org/project/9?task=183#bottom

Overall, I think looping through all points and doing your checks on `uniqueLats`, `uniqueLngs`, and `lngsForThisLat` will be enough. However, you want to do the checks after the whole loop finishes, so as to check against the "global picture" of the whole task boundary.

Reply:
Aha! In my spelunking around with various task manager tasks, I found some tasks that had extra points in the 'Points' associated with the FeatureCollection- these extra points were optional points with labels that displayed things like: 

![image](https://user-images.githubusercontent.com/1887955/66229849-c4afd100-e6b0-11e9-843f-5827126337b3.png)

That point is circled, and its lat/lng is not necessarily the same as the corner of the bbox.  This is why I elected to iterate through the lineString specifically and not just the generic list of points; those extra points would compromise our ability to determine rectangularity (note to self: make that a word). I hope that makes sense!

Reply:
Ah, sorry didn't see all this discussion until now. Yeah, thats an enhancement I added to our GPX boxes to add task info as points around the edges.